addons/frag/functions/fnc_dev_trackHitBox.sqf

```diff
@@ -0,0 +1,75 @@
+#include "..\script_component.hpp"
+/*
+ * Author: Lambda.Tiger
+ * Add a hit box outline to an object, outdated for unit hits as they
+ * use hit-point locations.
+ *
+ * Arguments:
+ * 0: Object that should have it's hit box drawn. <OBJECT>
+ * 1: Add sphere at object origin. <BOOL> (Default: true)
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [player] call ace_frag_fnc_dev_trackHitBox;
+ *
+ * Public: No
+ */
+
+params [
+    ["_object", objNull, [objNull]],
+    ["_addSphere", true, [true]]
+];
+TRACE_2("Adding hitbox",_object,_addSphere);
+
+if (isNull _object) exitWith {};
+
+// Grab the right hitBox
+private _box = [];
+if (_object isKindOf "CAManBase") then {
+    if (isNull objectParent _object) then {
+        _box = 0 boundingBox _object;
+    } else {
+        _box = boundingBoxReal [_object, "Geometry"];
+    };
+} else {
+    _box = boundingBoxReal [_object, "FireGeometry"];
+};
+_box params ["_lowP", "_upP"];
+
+// adjust with stance
+switch (stance _object) do {
+    case "STAND": {_upP set [2, 1.9];};
+    case "CROUCH": {_upP set [2, 1.3];};
+    case "PRONE": {_upP set [2, 0.8];};
+};
+private _centerPoint = ASLToAGL getPosASL _object;
+
+if (GVAR(dbgSphere) && {_addSphere && {isNull objectParent _object}}) then {
```

Comment:
Does it actually, or do you just believe it does?

Reply:
I tested it in game: If `GVAR(dbgSphere)` is false, then your code is slightly faster, otherwise mine is slightly faster.
Imo mine is more readable, as you don't have 2 brackets nested within each other, especially when it's not necessary.

Reply:
ok, sounds good

Reply:
done by https://github.com/acemod/ACE3/pull/9728/commits/66c566b772c9ff7f7a66a77e86f909b0fcab9f67